src/pages/Details.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import HelpLine from '../components/HelpLine';
+import MovieList from '../components/MovieList';
+
+import mockObject from '../MockFilmList';
+
+export default class Details extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.updateCurrentMovie();
```

Comment:
To store internal component data which affects view, use state object.

```
constructor(props) {
  super(props);
  this.state = {
    currentMovie: mockObject.find((el) => el.id === props.match.params.movieID)
  }
}
```

Reply:
Done